### PR TITLE
Ability to let the kubernetes decide node port

### PR DESCRIPTION
**Pull request by omer-dayan**

### CODE REVIEW

File: cmd/trainer/trainer_runai.go [hunk 1]
```diff
@@ -234,7 +234,42 @@ func (rt *RunaiTrainer) getRunaiTrainingJob(podSpecJob cmdTypes.PodTemplateJob,
...
-	return NewRunaiJob(filteredPods, lastCreatedPod, podSpecJob.CreationTimestamp, jobType, podSpecJob.Name, podSpecJob.Labels["app"] == "runaijob", []string{}, false, podSpecJob.Template.Spec, podSpecJob.Template.ObjectMeta, podSpecJob.ObjectMeta, podSpecJob.Namespace, ownerResource, status, podSpecJob.Parallelism, podSpecJob.Completions, podSpecJob.Failed, podSpecJob.Succeeded), nil
+	serviceUrls, err := rt.getServiceUrlsByLastCreatedPod(lastCreatedPod, namespace)
+	if err != nil {
+		return nil, err
+	}
+	return NewRunaiJob(filteredPods, lastCreatedPod, podSpecJob.CreationTimestamp, jobType, podSpecJob.Name, podSpecJob.Labels["app"] == "runaijob", serviceUrls, false, podSpecJob.Template.Spec, podSpecJob.Template.ObjectMeta, podSpecJob.ObjectMeta, podSpecJob.Namespace, ownerResource, status, podSpecJob.Parallelism, podSpecJob.Completions, podSpecJob.Failed, podSpecJob.Succeeded), nil
+}
+
+func (rt *RunaiTrainer) getServiceUrlsByLastCreatedPod(lastCreatedPod *v1.Pod, namespace string) ([]string, error) {
+	services, err := rt.getServicesInNamespace(namespace)
```
Comment (yodarshafrir1) [inline cmd/trainer/trainer_runai.go:245]: Can we make this method common with the there place who calls `getServicesInNamespace`?

File: cmd/trainer/trainer_runai.go [hunk 2]
```diff
@@ -505,22 +540,23 @@ func getExternalIPFromAnnotationHack(nodesList *v1.NodeList) string {
 func getServiceEndpoints(nodeIp string, service v1.Service) (urls []string) {
 	if service.Status.LoadBalancer.Ingress != nil && len(service.Status.LoadBalancer.Ingress) != 0 {
 		for _, port := range service.Spec.Ports {
+			fmt.Println(port)
```
Comment (yodarshafrir1) [inline cmd/trainer/trainer_runai.go:543]: Do we really want to print it?

File: cmd/trainer/trainer_runai.go [hunk 3]
```diff
@@ -234,7 +234,42 @@ func (rt *RunaiTrainer) getRunaiTrainingJob(podSpecJob cmdTypes.PodTemplateJob,
 
 	jobType := rt.getJobType(&podSpecJob)
 	status := getTrainingStatus(podSpecJob.ObjectMeta.Annotations, lastCreatedPod, podSpecJob.ExtraStatus)
-	return NewRunaiJob(filteredPods, lastCreatedPod, podSpecJob.CreationTimestamp, jobType, podSpecJob.Name, podSpecJob.Labels["app"] == "runaijob", []string{}, false, podSpecJob.Template.Spec, podSpecJob.Template.ObjectMeta, podSpecJob.ObjectMeta, podSpecJob.Namespace, ownerResource, status, podSpecJob.Parallelism, podSpecJob.Completions, podSpecJob.Failed, podSpecJob.Succeeded), nil
+	serviceUrls, err := rt.getServiceUrlsByLastCreatedPod(lastCreatedPod, namespace)
+	if err != nil {
+		return nil, err
+	}
+	return NewRunaiJob(filteredPods, lastCreatedPod, podSpecJob.CreationTimestamp, jobType, podSpecJob.Name, podSpecJob.Labels["app"] == "runaijob", serviceUrls, false, podSpecJob.Template.Spec, podSpecJob.Template.ObjectMeta, podSpecJob.ObjectMeta, podSpecJob.Namespace, ownerResource, status, podSpecJob.Parallelism, podSpecJob.Completions, podSpecJob.Failed, podSpecJob.Succeeded), nil
```
Comment (yodarshafrir1) [inline cmd/trainer/trainer_runai.go:241]: Did we not show the service urls before your change in the get command?

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]

File: cmd/trainer/trainer_runai.go (hunk 3 above)
Comment (omer-dayan) [reply cmd/trainer/trainer_runai.go:241]: No, we always showed the label, but put there empty list

**Review by yodarshafrir1 — APPROVED** [review]

**Pull request merged**